packages/@aws-cdk/aws-cognito/lib/user-pool.ts

```diff
@@ -850,52 +859,44 @@ export class UserPool extends Resource implements IUserPool {
   private schemaConfiguration(props: UserPoolProps): CfnUserPool.SchemaAttributeProperty[] | undefined {
     const schema: CfnUserPool.SchemaAttributeProperty[] = [];
 
-    if (props.requiredAttributes) {
-      const stdAttributes: StandardAttribute[] = [];
-
-      if (props.requiredAttributes.address) { stdAttributes.push(StandardAttribute.ADDRESS); }
-      if (props.requiredAttributes.birthdate) { stdAttributes.push(StandardAttribute.BIRTHDATE); }
-      if (props.requiredAttributes.email) { stdAttributes.push(StandardAttribute.EMAIL); }
-      if (props.requiredAttributes.familyName) { stdAttributes.push(StandardAttribute.FAMILY_NAME); }
-      if (props.requiredAttributes.fullname) { stdAttributes.push(StandardAttribute.NAME); }
-      if (props.requiredAttributes.gender) { stdAttributes.push(StandardAttribute.GENDER); }
-      if (props.requiredAttributes.givenName) { stdAttributes.push(StandardAttribute.GIVEN_NAME); }
-      if (props.requiredAttributes.lastUpdateTime) { stdAttributes.push(StandardAttribute.LAST_UPDATE_TIME); }
-      if (props.requiredAttributes.locale) { stdAttributes.push(StandardAttribute.LOCALE); }
-      if (props.requiredAttributes.middleName) { stdAttributes.push(StandardAttribute.MIDDLE_NAME); }
-      if (props.requiredAttributes.nickname) { stdAttributes.push(StandardAttribute.NICKNAME); }
-      if (props.requiredAttributes.phoneNumber) { stdAttributes.push(StandardAttribute.PHONE_NUMBER); }
-      if (props.requiredAttributes.preferredUsername) { stdAttributes.push(StandardAttribute.PREFERRED_USERNAME); }
-      if (props.requiredAttributes.profilePage) { stdAttributes.push(StandardAttribute.PROFILE_URL); }
-      if (props.requiredAttributes.profilePicture) { stdAttributes.push(StandardAttribute.PICTURE_URL); }
-      if (props.requiredAttributes.timezone) { stdAttributes.push(StandardAttribute.TIMEZONE); }
-      if (props.requiredAttributes.website) { stdAttributes.push(StandardAttribute.WEBSITE); }
-
-      schema.push(...stdAttributes.map((attr) => {
-        return { name: attr, required: true };
-      }));
+    if (props.standardAttributes) {
+      const standardAttributes = props.standardAttributes;
+      const stdAttributes = Object.keys(standardAttributes)
+        .filter(
+          attrName =>
+            !!standardAttributes[attrName as keyof StandardAttributes],
+        ).filter(
+          attrName =>
+            standardAttributes[attrName as keyof StandardAttributes]!.required ||
+            standardAttributes[attrName as keyof StandardAttributes]!.mutable,
```

Comment:
@ericzbeard tells me that from his experience, the combination of `required` and `mutable` (or maybe a different combination) is invalid, and the fact that it's invalid is not clear until the attribute change is attempted. And even at this point, the error message is generic/vague.

Perhaps we should add some validations to prevent users from doing this?

@ericzbeard - can you corroborate?

Reply:
I'm not totally clear on what this PR is doing. We should make it so that all of the regular attributes like given_name, etc are mutable by default, since the failure when using an IDP has no error message and is very hard to troubleshoot. (Is that the idea here?) What's not allowed is a required custom attribute.

Reply:
Ah I see. We're not dealing with custom attributes in this PR, so I guess that doesn't apply here.